cmake/Modules/FindTPLCUDA.cmake

```diff
@@ -7,37 +7,38 @@ IF (NOT CUDAToolkit_ROOT)
   ENDIF()
 ENDIF()
 
-# FIXME CMake 3.28.4 creates more targets than we export
-IF(CMAKE_VERSION VERSION_GREATER_EQUAL "3.17.0" AND CMAKE_VERSION VERSION_LESS "3.28.4")
-  find_package(CUDAToolkit)
-ELSE()
-  include(${CMAKE_CURRENT_LIST_DIR}/CudaToolkit.cmake)
-ENDIF()
-
-
-IF (TARGET CUDA::cudart)
-  SET(FOUND_CUDART TRUE)
-  KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cudart)
-ELSE()
-  SET(FOUND_CUDART FALSE)
+IF(KOKKOS_CXX_HOST_COMPILER_ID STREQUAL NVHPC AND CMAKE_VERSION VERSION_LESS "3.20.1"))
+  MESSAGE(FATAL_ERROR "Using NVHPC as host compiler requires at least CMake 3.20.1")
 ENDIF()
 
-IF (TARGET CUDA::cuda_driver)
-  SET(FOUND_CUDA_DRIVER TRUE)
-  KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cuda_driver)
-ELSE()
-  SET(FOUND_CUDA_DRIVER FALSE)
-ENDIF()
-
-include(FindPackageHandleStandardArgs)
-IF(KOKKOS_CXX_HOST_COMPILER_ID STREQUAL NVHPC)
-  SET(KOKKOS_CUDA_ERROR "Using NVHPC as host compiler requires at least CMake 3.20.1")
-ELSE()
-  SET(KOKKOS_CUDA_ERROR DEFAULT_MSG)
-ENDIF()
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(TPLCUDA ${KOKKOS_CUDA_ERROR} FOUND_CUDART FOUND_CUDA_DRIVER)
-IF (FOUND_CUDA_DRIVER AND FOUND_CUDART)
+IF(CMAKE_VERSION VERSION_GREATER_EQUAL "3.17.0")
+  find_package(CUDAToolkit REQUIRED)
   KOKKOS_CREATE_IMPORTED_TPL(CUDA INTERFACE
     LINK_LIBRARIES CUDA::cuda_driver CUDA::cudart
   )
+  KOKKOS_EXPORT_CMAKE_TPL(CUDAToolkit REQUIRED)
+ELSE()
+  include(${CMAKE_CURRENT_LIST_DIR}/CudaToolkit.cmake)
+
+  IF (TARGET CUDA::cudart)
+    SET(FOUND_CUDART TRUE)
+    KOKKOS_EXPORT_IMPORTED_TPL(CUDA::cudart)
+  ELSE()
+    SET(FOUND_CUDART FALSE)
+  ENDIF()
+
```

Comment:
Would a brief explanatory comment at the beginning of each of the different methods for finding CUDA be too obtrusive?  For instance, a comment for the block where it's assumed Kokkos will be in Trilinos, another for the case where Trilinos links to Kokkos as an external TPL, etc.?

Reply:
There is no different code for `Trilinos`. If the `CMake` version is too old (which is only slightly newer than our minimum) we use the file bundled with `Kokkos`, otherwise we use the one bundled with `CMake`.

I just wanted to make sure that this logic also works for `Trilinos`.